sw/host/rom_ext_image_tools/signer/image/src/image.rs

```diff
@@ -184,19 +184,102 @@ impl RawImage {
-            .expect("Failed to parse string to u64!"),
-        false => s.parse::<u64>().expect("Failed to parse string to u64!"),
-    };
+    let value = str_to_u64(s);
 
     if value <= (u32::MAX as u64) {
         (value as u32).to_le_bytes().to_vec()
     } else {
         value.to_le_bytes().to_vec()
     }
 }
+
+#[cfg(test)]
+mod tests {
+use super::*;
+
+use std::convert::TryInto;
+
+use rom_ext_config::parser::Extension;
+use rom_ext_config::parser::InputFiles;
+
+use std::env;
+
+    static DEV_RELATIVE_PATH: &str = "sw/host/rom_ext_image_tools/signer/dev";
+
+    #[test]
+    fn test_update_static_fields() {
+        let source_root = env::var("MESON_SOURCE_ROOT")
+            .expect("Failed, ENV variable not set!");
+
+        let dev_path = Path::new(&source_root).join(DEV_RELATIVE_PATH);
+        let full_path = |path| {
+            let mut new_path = PathBuf::new();
+            new_path.push(&dev_path);
+            new_path.push(path);
+            new_path
+        };
+
+        let config: ParsedConfig = ParsedConfig {
+            input_files: InputFiles {
+                image_path: full_path("rom_ext_blank_image.bin"),
+                private_key_der_path: PathBuf::from(""),
+                usage_constraints_path: full_path("usage_constraints.bin"),
+                system_state_value_path: PathBuf::from(""),
+            },
+            peripheral_lockdown_info: PeripheralLockdownInfo {
+                value: 0,
+            },
+            manifest_identifier: String::from("0x11111111"),
+            image_version: String::from("0x22222222"),
+            extensions: [
+                Extension {
+                    offset: String::from("0x33333333"),
+                    checksum: String::from("0x44444444"),
+                },
+                Extension {
+                    offset: String::from("0x55555555"),
+                    checksum: String::from("0x55555555"),
+                },
+                Extension {
+                    offset: String::from("0x66666666"),
+                    checksum: String::from("0x66666666"),
+                },
+                Extension {
+                    offset: String::from("0x77777777"),
+                    checksum: String::from("0x77777777"),
+                },
+            ],
+        };
+        let mut image = RawImage::new(&config.input_files.image_path);
+
+        // Make sure that the image is blank to start with.
+        for byte in image.data.iter() {
+            assert_eq!(*byte, 0);
+        }
+
+        image.update_static_fields(&config);
+
+        let check_u32 = |offset, value_as_str| {
+            let input_val = str_to_u64(value_as_str) as u32;
+            let read_val = u32::from_le_bytes(image.data[offset..4].try_into()
+                                              .unwrap()
+            );
+            assert_eq!(input_val, read_val);
+        };
+
+        check_u32(manifest::ROM_EXT_MANIFEST_IDENTIFIER_OFFSET as usize,
```

Comment:
This function is only used once...?

Reply:
Oh, I forgot to add other asserts! Thanks!

Reply:
Done.